### PR TITLE
Fixed CI Support with Python 3.12

### PR DESCRIPTION
**Pull request by Nusnus**

### CODE REVIEW

**Comment by auvipy** [comment]
good luck and good night for today. hopefully this will be fixed tomorrow, either by you or by me or by us.

**Comment by Nusnus** [comment]
> good luck and good night for today. hopefully this will be fixed tomorrow, either by you or by me or by us.

Awesome job today @auvipy !!
I'll try to see what's going on here, thanks!

**Comment by auvipy** [comment]
I will return to this tomorrow as well.

**Comment by Nusnus** [comment]
@auvipy ready for review & merge :)

File: t/unit/tasks/test_result.py [hunk 1]
```diff
@@ -389,12 +390,17 @@ def test_ready(self):
 
         assert not self.app.AsyncResult(uuid()).ready()
 
+    @pytest.mark.skipif(
+        platform.python_implementation() == "PyPy",
+        reason="Mocking here doesn't play well with PyPy",
```
Comment (Nusnus) [inline t/unit/tasks/test_result.py:395]: TBH I'm not sure why it fails only with PyPy, but I skip it to allow us to move forward.
Not sure if it's acceptable @auvipy

Comment (auvipy) [reply t/unit/tasks/test_result.py:395]: I am goin to check it in  my work hours today.

**Review by auvipy — APPROVED** [review]
I think we can skip PyPy for this as the sole target of this release is to have python 3.12 compatibility!

And there are some back ends which do not play well with pypy as well. so we did skip in past as well. thanks for taking the time to figure it out! <3

File: .github/workflows/python-package.yml [hunk 2]
```diff
@@ -75,7 +75,7 @@ jobs:
       strategy:
           fail-fast: false
           matrix:
-              python-version: ['3.8', '3.9', '3.10', '3.11']
+              python-version: ['3.8', '3.9', '3.10', '3.11', '3.12']
```
Comment (auvipy) [inline .github/workflows/python-package.yml:78]: I missed it

**Comment by Nusnus** [comment]
> thanks for taking the time to figure it out! <3

Team Effort 💪

**Pull request merged**